src/components/Comments/ReplyCommentItem.js

```diff
@@ -282,14 +282,16 @@ const ReplyCommentItem = ({
             </ButtonHightlight>
           </TouchableOpacity>
         )}
-        <TouchableOpacity onPress={() => onBlock(comment)} testID="btnBlock" activeOpacity={1}>
-          <ButtonHightlight
-            onLongPress={handleLongPress}
-            onPress={() => onBlock(comment)}
-            style={[styles.btnBlock(comment.user.id === yourselfId), styles.btn]}>
-            <IconEn name="block" size={15.02} color={COLORS.balance_gray} />
-          </ButtonHightlight>
-        </TouchableOpacity>
+        {!comment?.is_you && (
+          <TouchableOpacity onPress={() => onBlock(comment)} testID="btnBlock" activeOpacity={1}>
+            <ButtonHightlight
+              onLongPress={handleLongPress}
+              onPress={() => onBlock(comment)}
+              style={[styles.btnBlock(comment.user.id === yourselfId), styles.btn]}>
+              <IconEn name="block" size={15.02} color={COLORS.balance_gray} />
+            </ButtonHightlight>
+          </TouchableOpacity>
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for hiding the block button if the comment is from the current user seems correct. However, it would be better to add a null check for `comment.user` before accessing its `id` property in line 290. This will prevent potential null reference errors.
```diff
- style={[styles.btnBlock(comment.user.id === yourselfId), styles.btn]}>
+ style={[styles.btnBlock(comment?.user?.id === yourselfId), styles.btn]}>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->